src/backend/distributed/commands/alter_table.c

```diff
@@ -1183,6 +1196,72 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 }
 
 
+/*
+ * AnyColumnsRequireDataCopy decides if relation has any columns that we
+ * might need to copy the data of when replacing table.
+ */
+static bool
+AnyColumnsRequireDataCopy(Oid relationId)
+{
+	return list_length(GetNonGeneratedStoredColumnNameList(relationId)) > 0;
+}
+
+
+/*
+ * BuildInsertColumnStringForReplaceTable is a helper function that returns
+ * comma seperated columns that we can insert into for ReplaceTable.
+ */
+static char *
+BuildInsertColumnStringForReplaceTable(Oid relationId)
+{
+	/*
+	 * When copying data to temp table, we skip columns having GENERATED
+	 * GENERATED ALWAYS AS (...) STORED expressions since Postgres doesn't
+	 * allow inserting into such columns. This is not bad since Postgres
+	 * would already generate such columns.
+	 */
+	List *nonStoredColumnNameList = GetNonGeneratedStoredColumnNameList(relationId);
+	return StringJoin(nonStoredColumnNameList, ',');
+}
+
+
+/*
+ * GetNonGeneratedStoredColumnNameList returns a list of column names for
+ * columns not having GENERATED ALWAYS AS (...) STORED expressions.
+ */
+static List *
+GetNonGeneratedStoredColumnNameList(Oid relationId)
+{
+	List *nonStoredColumnNameList = NIL;
+
+	Relation relation = relation_open(relationId, AccessShareLock);
+	TupleDesc tupleDescriptor = RelationGetDescr(relation);
+	for (int columnIndex = 0; columnIndex < tupleDescriptor->natts; columnIndex++)
+	{
+		Form_pg_attribute currentColumn = TupleDescAttr(tupleDescriptor, columnIndex);
+		if (currentColumn->attisdropped)
+		{
+			continue;
+		}
+
+#if PG_VERSION_NUM >= 120000
+		if (currentColumn->attgenerated == ATTRIBUTE_GENERATED_STORED)
```

Comment:
What about `ATTRIBUTE_IDENTITY_ALWAYS ` and `ATTRIBUTE_IDENTITY_BY_DEFAULT`?

Reply:
https://github.com/citusdata/citus/pull/4616#discussion_r568524393

Reply:
Let's add comment on the two others must be tackled separately, and open a new issue to track?

Reply:
Yeah, I will open an issue for that.
Other than that, not sure if it's a good place to add such comment since even if we supported such identity columns, we wouldn't have to handle such columns when executing `INSERT .. SELECT` here, what do you think ?

Reply:
Yes, makes sense

Reply:
I will open that issue after opening pr's to error out for identity cols in `CreateCitusLocalTable` & `ReplaceTable`